pypi-setuptools.yaml

```diff
@@ -0,0 +1,60 @@
+package:
+  name: pypi-setuptools
+  version: 69.2.0
+  epoch: 1
+  description: Easily download, build, install, upgrade, and uninstall Python packages
+  copyright:
+    - license: "MIT"
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - python-3.10-base
+      - python-3.11-base
+      - python-3.12-base
+      - wolfi-base
+  environment:
+    # This is needed to work around the error "ValueError: ZIP does not support timestamps before 1980"
+    SOURCE_DATE_EPOCH: 315532800
+
+vars:
+  pypi-package: setuptools
+
+data:
+  - name: py-versions
+    items:
+      3.12: ""
+      3.11: ""
+      3.10: ""
+
+pipeline:
+  - uses: fetch
+    with:
+      expected-sha256: 0ff4183f8f42cd8fa3acea16c45205521a4ef28f73c6391d8a25e92893134f2e
+      uri: https://files.pythonhosted.org/packages/source/s/setuptools/setuptools-${{package.version}}.tar.gz
+
+  - runs: mkdir -p ${{targets.destdir}}
+
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: Easily download, build, install, upgrade, and uninstall Python packages
+    pipeline:
+      - runs: |
+          python${{range.key}} setup.py install --no-compile --root="${{targets.subpkgdir}}" --prefix=/usr
+          rm -rf build/
+          python${{range.key}} -m compileall --invalidation-mode=unchecked-hash -r100 ${{targets.subpkgdir}}/usr/lib/
```

Comment:
fwiw, i am not convinced i like 'unchecked-hash', i ran into an issue with it today ... admittedly debugging, but if you go and change a .py file it will _not_ change code path. the .pyc will be used and run and a stack trace will show the actual line number (from the .pyc file) but the content of the  .py file. 

lets just say its confusing.

Reply:
@smoser the other two options are force hash check (where on startup all .py files are read and hashed), or switch to timestamp (where on startup all .py files are stat'ed and timestamps are compared). In practice, most other distributions do timestamp check (as they generate pyc files in maintainer scripts).